src/ParameterResolver/TypeHintResolver.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Invoker\ParameterResolver;
+
+use Invoker\ParameterResolver\ParameterResolver;
+use ReflectionFunctionAbstract;
+
+/**
+ * Inject entries using type-hints.
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class TypeHintResolver implements ParameterResolver
+{
+    public function getParameters(
+        ReflectionFunctionAbstract $reflection,
+        array $providedParameters,
+        array $resolvedParameters
+    ) {
+        $parameters = $reflection->getParameters();
+
+        // Skip parameters already resolved
+        if (! empty($resolvedParameters)) {
+            $parameters = array_diff_key($parameters, $resolvedParameters);
+        }
+
+        foreach ($parameters as $index => $parameter) {
+            $parameterClass = $parameter->getClass();
+
+            if ($parameterClass && array_key_exists($parameterClass->name, $providedParameters)) {
```

Comment:
`isset($providedParameters[$parameterClass->name])` could work here because the class name will not be null (micro-optimization, but might as well do it because this will be called a lot)